x-pack/plugins/cases/server/services/user_profiles/index.ts

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import Boom from '@hapi/boom';
+import { pipe } from 'fp-ts/lib/pipeable';
+import { fold } from 'fp-ts/lib/Either';
+import { identity } from 'fp-ts/lib/function';
+
+import { KibanaRequest, Logger } from '@kbn/core/server';
+import {
+  SecurityPluginSetup,
+  SecurityPluginStart,
+  UserProfileServiceStart,
+} from '@kbn/security-plugin/server';
+import { SpacesPluginStart } from '@kbn/spaces-plugin/server';
+
+import { excess, SuggestUserProfilesRequestRt, throwErrors } from '../../../common/api';
+import { Operations } from '../../authorization';
+import { createCaseError } from '../../common/error';
+
+interface UserProfileOptions {
+  securityPluginSetup?: SecurityPluginSetup;
+  securityPluginStart?: SecurityPluginStart;
```

Comment:
Yeah I found that the security plugin can still be disabled. The functionality to disable it was moved into Elasticsearch. When we initially developed the cases RBAC we intentionally allowed the security plugin to be disabled. It was something that Larry suggested we support. So I didn't want to deviate from that in this PR. I think some users want to play around with Kibana without security enable and it'd be nice for them to still have access to cases in that situation.

Reply:
> Yeah I found that the security plugin can still be disabled. The functionality to disable it was moved into Elasticsearch. 

Yeah, but you'll still get security contract even if Security is disabled in Elasticsearch and can replace `securityPluginStart?: SecurityPluginStart;` with `  securityPluginStart: SecurityPluginStart;` to have less "undefined" checks (since you need to rely on the availabilty of the security feature via license check anyway). But it's minor, just wanted to point this out.

Reply:
ah ok, good to know. I'll create an issue to fix that up

Reply:
https://github.com/elastic/kibana/issues/137861